modules/Users/Login.php

```diff
@@ -203,25 +205,28 @@ function validateAndSubmit(){
 		 	<td colspan='2'><div style='margin-left:2px'class='x-sqs-list' id='recaptcha_image'></div></td>
 		</tr>
 		<tr>
-			<td colspan='2' align='right'><a href='javascript:Recaptcha.reload()'>".$mod_strings['LBL_RECAPTCHA_NEW_CAPTCHA']."</a>&nbsp;&nbsp;
-			 		<a class='recaptcha_only_if_image' href='javascript:Recaptcha.switch_type(\"audio\")'>".$mod_strings['LBL_RECAPTCHA_SOUND']."</a>
-			 		<a class='recaptcha_only_if_audio' href='javascript:Recaptcha.switch_type(\"image\")'> ".$mod_strings['LBL_RECAPTCHA_IMAGE']."</a>
+			<td colspan='2' align='right'><a href='javascript:Recaptcha.reload()'>" . $mod_strings['LBL_RECAPTCHA_NEW_CAPTCHA'] . "</a>&nbsp;&nbsp;
+			 		<a class='recaptcha_only_if_image' href='javascript:Recaptcha.switch_type(\"audio\")'>" . $mod_strings['LBL_RECAPTCHA_SOUND'] . "</a>
+			 		<a class='recaptcha_only_if_audio' href='javascript:Recaptcha.switch_type(\"image\")'> " . $mod_strings['LBL_RECAPTCHA_IMAGE'] . "</a>
 		 	</td>
 		</tr>";
-		$sugar_smarty->assign('CAPTCHA', $Captcha);
-		echo $captcha_js;
+    $sugar_smarty->assign('CAPTCHA', $Captcha);
+    echo $captcha_js;
 
-	}else{
-		echo "<script>
+} else {
+    echo "<script>
 		function validateAndSubmit(){generatepwd();}
 		</script>";
-	}
+}
 
-if (file_exists('themes/'.SugarThemeRegistry::current().'/tpls/login.tpl')) {
-	echo $sugar_smarty->display('themes/'.SugarThemeRegistry::current().'/tpls/login.tpl');
+if (file_exists('custom/themes/' . SugarThemeRegistry::current() . '/login.tpl')) {
+    echo $sugar_smarty->display('custom/themes/' . SugarThemeRegistry::current() . '/login.tpl');
 } elseif (file_exists('custom/modules/Users/login.tpl')) {
-	echo $sugar_smarty->display('custom/modules/Users/login.tpl');
+    echo $sugar_smarty->display('custom/modules/Users/login.tpl');
+} elseif (file_exists('themes/' . SugarThemeRegistry::current() . '/login.tpl')) {
+    echo $sugar_smarty->display('themes/' . SugarThemeRegistry::current() . '/login.tpl');
+} elseif (file_exists('modules/Users/login.tpl')) {
+    echo $sugar_smarty->display('modules/Users/login.tpl');
 } else {
-	echo $sugar_smarty->display('modules/Users/login.tpl');
+    $GLOBALS['log']->fatal('login.tpl not found');
```

Comment:
To ensure that this gets picked up by the developer. You should throw an exception with the same message. This will then be output to the suitecrm.log and to the apache error log but with a stack trace.

Example:
`throw new Exception('login.tpl not found');`

To ensure that the user is aware that an error has occured. You should add a label into include/language/en_us.lang.php and echo the label out to the page

Example:
`echo <span class="error="> $app_strings['LBL_MISSING_TEMPLATE']</span>`